contrib/dimfeld/httptreemux/v5/httptreemux.go

```diff
@@ -42,7 +42,41 @@ func New(opts ...RouterOption) *Router {
 
 // ServeHTTP implements http.Handler.
 func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	resource := r.config.resourceNamer(r, w, req)
+	resource := r.config.resourceNamer(r.TreeMux, w, req)
+	// pass r.TreeMux to avoid a circular reference panic on calling r.ServeHTTP
+	httptrace.TraceAndServe(r.TreeMux, w, req, &httptrace.ServeConfig{
+		Service:  r.config.serviceName,
+		Resource: resource,
+		SpanOpts: r.config.spanOpts,
+	})
+}
+
+// ContextRouter is a traced version of httptreemux.ContextMux.
+type ContextRouter struct {
+	*httptreemux.ContextMux
+	config *routerConfig
+}
+
+// NewWithContext returns a new router augmented with tracing and preconfigured
+// to work with context objects. The matched route and parameters are added to
+// the context.
+func NewWithContext(opts ...RouterOption) *ContextRouter {
```

Comment:
My options were either this or `NewContextRouter` to be similar to [httptreemux.NewContextMux](https://github.com/dimfeld/httptreemux/blob/master/treemux_17.go#L105-L113).

Reply:
I confirmed with the reviewers that the naming is good to go. :tada: